docs/proposals/local-cluster.md

```diff
@@ -0,0 +1,486 @@
+After this command the user can access the cluster conveniently
+by issuing the following additional commands:
+```shell
+export KUBECONFIG=$HOME/.local-cluster/admin.conf
+alias kubectl=$HOME/.local-cluster/kubectl
+```
+For this command to work with `BUILD_*` options, it must be invoked
+from Kubernetes source directory.
+
+```shell
+./local-cluster.sh down
+```
+If the cluster is up, brings it down, deleting all the node
+containers, but keeping the current cluster snapshot. This command
+does nothing if the cluster is not running.
+
+```shell
+./local-cluster.sh clean
+```
+Same as `./local-cluster.sh down`, but also deletes the current
+cluster snapshot if it exists, along with `k8s-local-cluster` docker
+network.
+
+```shell
+./local-cluster.sh e2e
+./local-cluster.sh e2e "test name substring"
+```
+When invoked without extra arguments, this runs all the non-serial e2e
+conformance tests against the current cluster. If an argument is
+given, the command will run e2e tests that have the specified
+substring in their name. When using this command, `GINKGO_PARALLEL` is
+set to a non-empty string so the tests run in parallel.
+
+```shell
+./local-cluster.sh e2e-serial
+./local-cluster.sh e2e-serial "test name substring"
+```
+When invoked without extra arguments, this runs all the serial e2e
+conformance tests against the current cluster. If an argument is
+given, the command will run e2e tests that have the specified
+substring in their name. When using this command the tests are not run
+in parallel.
+
+```shell
+./local-cluster.sh reup [flags...]
+```
+Same as `./local-cluster.sh up`, but also rebuilds hyperkube & kubectl
+binaries and updates them in case if Kubernetes is built from source.
+This command can still reuse the last snapshot.
+
+## Configuration variables
+
+The variables are listed with their default values and
+corresponding command line flags.
+
+`NUM_NODES` (`--num-nodes`, defaults to 2) specifies the number of
+worker nodes to create. If set to `0`, only master node is set up and
+`dedicated` taint is removed from it.
+
+`CNI_PLUGIN` (`--cni-plugin`, defaults to `bridge`) specifies CNI
+plugin to use. When `bridge` plugin is used, `local-cluster` joins the
+bridges from the nodes together with another bridge. Other supported
+CNI implementations are `weave`, `flannel` and `calico`.
+
+`NO_DEDICATED` (`--no-dedicated`, not set by default) removes
+`dedicated` taint from master node even if `NUM_NODES` is greater than
+zero.
+
+`DIND_SUBNET` (`--dind-subnet`, default: `10.192.0.0/16`) specifies a
+CIDR notation IP range for use by node containers.
+
+`APISERVER_PORT` (`--apiserver-port`, default: `8080`) specifies
+apiserver port to use. If you're working with remote Docker engine,
+you need to forward this port to `localhost` over ssh.
+
+`BUILD_KUBEADM` (`--build-kubeadm`, not set by default) instructs
+`local-cluster` to build `kubeadm` locally from Kubernetes sources.
```

Comment:
It defaults to current directory (will document this) but I'll also add an option to specify it.